lib/matplotlib/axis.py

```diff
@@ -1051,60 +1051,28 @@ def _update_ticks(self, renderer):
-        # outside of the defined axis interval.
-        if interval[0] <= interval[1]:
-            interval_expanded = interval
-        else:
-            interval_expanded = interval[1], interval[0]
-
-        if hasattr(self, '_get_pixel_distance_along_axis'):
-            # normally, one does not want to catch all exceptions that
-            # could possibly happen, but it is not clear exactly what
-            # exceptions might arise from a user's projection (their
-            # rendition of the Axis object).  So, we catch all, with
-            # the idea that one would rather potentially lose a tick
-            # from one side of the axis or another, rather than see a
-            # stack trace.
-            # We also catch users warnings here. These are the result of
-            # invalid numpy calculations that may be the result of out of
-            # bounds on axis with finite allowed intervals such as geo
-            # projections i.e. Mollweide.
-            with np.errstate(invalid='ignore'):
-                try:
-                    ds1 = self._get_pixel_distance_along_axis(
-                        interval_expanded[0], -0.5)
-                except Exception:
-                    warnings.warn("Unable to find pixel distance along axis "
-                                  "for interval padding of ticks; assuming no "
-                                  "interval padding needed.")
-                    ds1 = 0.0
-                if np.isnan(ds1):
-                    ds1 = 0.0
-                try:
-                    ds2 = self._get_pixel_distance_along_axis(
-                        interval_expanded[1], +0.5)
-                except Exception:
-                    warnings.warn("Unable to find pixel distance along axis "
-                                  "for interval padding of ticks; assuming no "
-                                  "interval padding needed.")
-                    ds2 = 0.0
-                if np.isnan(ds2):
-                    ds2 = 0.0
-            interval_expanded = (interval_expanded[0] - ds1,
-                                 interval_expanded[1] + ds2)
+        if interval[1] <= interval[0]:
+             interval = interval[1], interval[0]
+        inter = self.get_transform().transform(interval)
 
         ticks_to_draw = []
         for tick, loc, label in tick_tups:
+            # draw each tick if it is in interval.   Note the transform
+            # to pixel space to take care of log transforms etc.
+            # interval_contains has a floating point tolerance.
             if tick is None:
                 continue
             # NB: always update labels and position to avoid issues like #9397
             tick.update_position(loc)
             tick.set_label1(label)
             tick.set_label2(label)
-            if not mtransforms.interval_contains(interval_expanded, loc):
+            try:
+                loct = self.get_transform().transform(loc)
```

Comment:
Not sure if it really makes a performance difference, but AFAICS `transform()` works on numpy arrays. So this could be pulled out of the loop and calculated in one go. Well, except for the `AssertionError` - when does that occur?

Reply:
We still have to do the loop, so can probably keep as-is for now.  

I think the Assertion Error happened in CI, possibly get_transform returned None for some wacky axis.

Reply:
Yeah its from the doc build: `gallery/scales/custom_scale.py`.  

```
Exception occurred:
  File "/home/circleci/project/lib/matplotlib/transforms.py", line 1436, in transform
    assert not np.ma.is_masked(res)  # just to be on the safe side
AssertionError
```

The custom scale uses masks, and transforms.transform doesn't like it.  

The previous implementation didn't call the transform, but this one does so that we can do the tolerance checking in log-scale rather than data space.

Reply:
Are we sure that there are still any ticks on that example when we hit this?